src/components/MutationsTable.tsx

```diff
@@ -0,0 +1,257 @@
+  locationSelector: LocationSelector;
+  baselineVariant?: VariantSelector;
+  dateRangeSelector: DateRangeSelector;
+};
+
+export const BaselineMutationsTable = ({
+  baselineData,
+  mutations,
+  mutationType,
+  locationSelector,
+  baselineVariant,
+  dateRangeSelector,
+}: BaselineMutationsTableProps) => {
+  const headers: string[] = ['Additional', 'Missing'];
+  console.log('baseline data', baselineData);
+
+  return (
+    <>
+      <div className='mb-4 italic'>
+        *The table compares the mutation set of the variants in the collection to the fixed /{'>'}90%
+        mutations of the baseline.
+      </div>
+      <div className='table-container'>
+        <table className='mutationCustomTable' data-testid='baseline-mutation-table'>
+          <thead>
+            <tr>
+              <td> </td>
+              {headers.map(header => (
+                <td key={header} colSpan={4}>
+                  {header}
+                </td>
+              ))}
+            </tr>
+            <tr>
+              <td>Name</td>
+              {columnNames.map((name, i) => (
+                <td key={`${name}-${i}`}>{name}</td>
+              ))}
+              {columnNames.map((name, i) => (
+                <td key={`${name}/${i}`}>{name}</td>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            {mutations &&
+              mutations.map(item => (
+                <tr key={item.name}>
+                  <td>
+                    {renderNameCell(
+                      item.query,
+                      locationSelector,
+                      item.name,
+                      true,
+                      dateRangeSelector,
+                      baselineVariant
+                    )}
+                  </td>
+                  {mutationColumnsProperties.map(field => {
+                    return (
+                      <td key={field.label}>
+                        {mutationType === 'aa'
+                          ? sortMutationList(
+                              'aa',
+                              pullAll(
+                                getNames(filterMutations(item.aaMutations, field)),
+                                getNames(filterBaselineMutations(baselineData.aaMutations))
+                              )
+                            ).join(', ')
+                          : sortMutationList(
+                              'nuc',
+                              pullAll(
+                                getNames(filterMutations(item.nucMutations, field)),
+                                getNames(filterBaselineMutations(baselineData.nucMutations))
+                              )
+                            ).join(', ')}
+                      </td>
+                    );
+                  })}
+
+                  {mutationColumnsProperties.map(field => {
+                    return (
+                      <td key={field.min}>
+                        {mutationType === 'aa'
```

Comment:
There is a lot of duplication in those places. It always follows the same pattern:
`mutationType === 'aa' ? doSomething(item.aaMutations) : doSomething(item.nucMutations)`

So the code does exactly the same (as far as I can judge), except for "either get aa or nuc mutations" and "sort according to aa or nuc".

Could we generalize the code a bit to reduce duplication? Introduce a clcass that wraps around mutationType and the mutation data? But there is no need to do that if it's too much work.